MatrixSDK/Data/MXEventTimeline.m

```diff
@@ -125,16 +137,58 @@ - (void)resetPagination
     [store resetPaginationOfRoom:_state.roomId];
 }
 
+- (MXHTTPOperation *)resetPaginationAroundInitialEventWithLimit:(NSUInteger)limit success:(void (^)())success failure:(void (^)(NSError *))failure
+{
+    NSAssert(_initialEventId, @"[MXEventTimeline] resetPaginationAroundInitialEventWithLimit cannot be called on live timeline");
+
+    // Reset the store
+    [store deleteAllData];
+
+    forwardsPaginationToken = nil;
+    hasReachedHomeServerForwardsPaginationEnd = NO;
+
+    // Get the context around the initial event
+    return [room.mxSession.matrixRestClient contextOfEvent:_initialEventId inRoom:room.roomId limit:limit success:^(MXEventContext *eventContext) {
+
+        // And fill the timelime with received data
+        [self initialiseState:eventContext.state];
+
+        // Reset pagination state from here
+        [self resetPagination];
+
+        [self addEvent:eventContext.event direction:MXTimelineDirectionForwards fromStore:NO];
+
+        for (MXEvent *event in eventContext.eventsBefore)
+        {
+            [self addEvent:event direction:MXTimelineDirectionBackwards fromStore:NO];
+        }
+
+        for (MXEvent *event in eventContext.eventsAfter)
+        {
+            [self addEvent:event direction:MXTimelineDirectionForwards fromStore:NO];
+        }
+
+        [store storePaginationTokenOfRoom:room.roomId andToken:eventContext.start];
+        forwardsPaginationToken = eventContext.end;
+
+        success();
```

Comment:
I can add a NSParameterAssert to check its existence. It is mandatory as the sdk user can do nothing while the request is not comple.